packages/bull-vault/test/fuzz-test/FlashBullFuzzTest.t.sol

```diff
@@ -0,0 +1,340 @@
+    address internal eulerMarketsModule;
+    address internal eToken;
+    address internal dToken;
+    address internal wPowerPerp;
+    address internal ethWSqueethPool;
+    address internal ethUsdcPool;
+
+    uint256 internal user1Pk;
+    uint256 internal deployerPk;
+    address internal user1;
+    address internal deployer;
+
+    uint256 internal cap;
+
+    // var to avoid stack too deep in test functions
+    uint256 userEthBalanceBeforeTx;
+    uint256 bullToMint;
+
+    function setUp() public {
+        string memory FORK_URL = vm.envString("FORK_URL");
+        vm.createSelectFork(FORK_URL, 15781550);
+
+        deployerPk = 0xAB11CE;
+        deployer = vm.addr(deployerPk);
+
+        vm.startPrank(deployer);
+        euler = 0x27182842E098f60e3D576794A5bFFb0777E025d3;
+        eulerMarketsModule = 0x3520d5a913427E6F0D6A83E07ccD4A4da316e4d3;
+        controller = Controller(0x64187ae08781B09368e6253F9E94951243A493D5);
+        crabV2 = CrabStrategyV2(0x3B960E47784150F5a63777201ee2B15253D713e8);
+        bullStrategy = new BullStrategy(
+            deployer,
+            address(crabV2),
+            address(controller),
+            euler,
+            eulerMarketsModule
+        );
+        flashBull = new FlashBull(
+            address(bullStrategy),
+            0x1F98431c8aD98523631AE4a59f267346ea31F984
+        );
+        usdc = controller.quoteCurrency();
+        weth = controller.weth();
+        eToken = IEulerMarkets(eulerMarketsModule).underlyingToEToken(weth);
+        dToken = IEulerMarkets(eulerMarketsModule).underlyingToDToken(usdc);
+        wPowerPerp = controller.wPowerPerp();
+        ethWSqueethPool = IController(bullStrategy.powerTokenController()).wPowerPerpPool();
+        ethUsdcPool = IController(bullStrategy.powerTokenController()).ethQuoteCurrencyPool();
+        testUtil = new TestUtil(
+            address(bullStrategy),
+            address(controller),
+            eToken,
+            dToken,
+            address(crabV2)
+        );
+
+        cap = 100000e18;
+        bullStrategy.setCap(cap);
+
+        vm.stopPrank();
+        user1Pk = 0xA11CE;
+        user1 = vm.addr(user1Pk);
+
+        vm.label(user1, "User 1");
+        vm.label(address(bullStrategy), "BullStrategy");
+        vm.label(address(flashBull), "FlashBull");
+        vm.label(euler, "Euler");
+        vm.label(eulerMarketsModule, "EulerMarkets");
+        vm.label(usdc, "USDC");
+        vm.label(weth, "WETH");
+
+        vm.deal(user1, 100000000e18);
+        // this is a crab whale, get some crab token from
+        vm.prank(0x06CECFbac34101aE41C88EbC2450f8602b3d164b);
+        IERC20(crabV2).transfer(user1, 100e18);
+        // some WETH and USDC rich address
+        vm.prank(0x57757E3D981446D585Af0D9Ae4d7DF6D64647806);
+        IERC20(weth).transfer(user1, 10100e18);
+    }
+
+    function testFuzzingFlashDeposit(uint256 _ethToCrab) public {
+        _ethToCrab = bound(_ethToCrab, 1e16, 600e18);
```

Comment:
why 600 upper bound?

Reply:
I don't know, will look at it later, this is outside of this PR scope